crates/adapters/okx/src/http/client.rs

```diff
@@ -902,21 +940,35 @@ impl OKXHttpClient {
                 false
             };
 
-            if let Some(l) = limit {
-                let max_limit = if use_history_endpoint { 100 } else { 300 };
-                let effective_limit = l.min(max_limit);
-                tracing::debug!(
-                    "Requested limit: {}, effective limit: {}, endpoint: {}",
-                    l,
-                    effective_limit,
-                    if use_history_endpoint {
-                        "history"
-                    } else {
-                        "regular"
+            // Apply endpoint-specific limit validation
+            let effective_limit = if let Some(l) = limit {
```

Comment:
I think we can use the `before` and `after` query params to paginate to any reasonable time range requested? We don't have to cap at the default of 100.